### ISSUE TITLE
oxfmt: standalone binary silently excludes all files on aarch64 Linux (gnu and musl)

### ISSUE THREAD

**Issue by wberry-godaddy** [oxc-project/oxc#23655]
## Summary

The `oxfmt` standalone binary (`aarch64-unknown-linux-gnu` and `aarch64-unknown-linux-musl`) silently excludes all target files on aarch64 Linux, regardless of:
- Working directory
- Whether a `.git` repo is present
- Whether ignore files exist
- Whether files are passed explicitly by path (relative or absolute)
- Whether `--ignore-path` is specified

The only file ever processed is `.oxfmtrc.jsonc` when present, because it appears to be handled as a special case (the config file) rather than going through the normal file walker.

The npm package (`oxfmt` via NAPI binding `@oxfmt/binding-linux-arm64-gnu`) does **not** have this issue and works correctly on the same hardware.

---

## Reproduction

Minimal reproduction in a fresh Ubuntu 22.04 arm64 Docker container:

```bash
docker run --rm --platform linux/arm64 ubuntu:22.04 bash -c '
  apt-get install -y -qq curl
  curl -sL https://github.com/oxc-project/oxc/releases/download/apps_v1.70.0/oxfmt-aarch64-unknown-linux-gnu.tar.gz     | tar -xz -C /usr/local/bin
  mv /usr/local/bin/oxfmt-aarch64-unknown-linux-gnu /usr/local/bin/oxfmt

  # Minimal test: fresh directory, no git, no ignore files, single file
  mkdir /test && echo "# hello" > /test/README.md
  cd /test && oxfmt --check README.md
'
```

**Output:**
```
Expected at least one target file. All matched files may have been excluded by ignore rules.
```

The same test with the npm package succeeds:

```bash
docker run --rm --platform linux/arm64 node:24-slim bash -c '
  mkdir /test && echo "# hello" > /test/README.md
  cd /test && npx oxfmt@0.55.0 --check README.md
'
```

**Output:**
```
All matched files use the correct format.
Finished in 73ms on 1 files using 1 threads.
```

---

## Additional context

- Tested on AWS Graviton (`c7g.medium` instances) running Ubuntu 22.04 in GitHub Actions self-hosted runners — same result
- Affects both `aarch64-unknown-linux-gnu` and `aarch64-unknown-linux-musl` builds
- Reproduced across releases v0.50.0, v0.54.0, and v0.55.0 — not a recent regression
- `--ignore-path /dev/null`, `--ignore-path <empty-file>`, and absolute paths all have the same result
- The npm NAPI binding works correctly on the same hardware, suggesting the file walker issue is specific to the standalone binary build

## Environment

- oxfmt version: `0.55.0`
- OS: Ubuntu 22.04 (`linux/arm64`)
- Binary: `aarch64-unknown-linux-gnu` (also reproduced with `aarch64-unknown-linux-musl`)

**Comment by wberry-godaddy** [comment]
The AGENTS.md describes the pure Rust binary as having a "Minimum feature set, CLI usage only, no LSP, no Stdin support — formats JS/TS, JSON and TOML files, no xxx-in-js support." This implies the current behaviour (silently skipping YAML/Markdown) is intentional design. The bug is that the error message — "Expected at least one target file. All matched files may have been excluded by ignore rules" — is deeply misleading when the real reason is "this build doesn't support these file types."

Should this project be forcing users to use the npm package when a binary (with more features enabled) could/should suffice?

**Comment by leaysgur** [comment]
For the time being, we recommend using the npm version.
(However, since we want to reduce our dependence on Prettier, we hope to expand the number of supported languages in the Rust version as well, so that line in the sand may change eventually.)

Also, we would like to note that clear in our documentation.
https://github.com/oxc-project/website/issues/1125
